### ISSUE TITLE
TLAPM Crash

### ISSUE THREAD

**Issue by kape1395** [tlaplus/tlapm#8]
```
Error running tlapm. Report a bug with the error code to the developers at https://github.com/tlaplus/tlapm/issues.
Error code: 3
```
The stacktrace is:
```
Oops, this seems to be a bug in TLAPM.
Please give feedback to developers.


 File "errors.ml", line 75, characters 2-8: Assertion failed
Raised at file "errors.ml", line 75, characters 2-15
Called from file "e_constness.ml", line 75, characters 23-75
Called from file "e_visit.ml", line 64, characters 15-24
Called from file "e_constness.ml", line 62, characters 16-32
Called from file "e_visit.ml", line 215, characters 22-31
Called from file "e_visit.ml", line 221, characters 23-31
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 222, characters 24-33
Called from file "e_visit.ml", line 145, characters 22-31
Called from file "e_constness.ml", line 44, characters 22-42
Called from file "action.ml", line 237, characters 12-24
Called from file "action.ml", line 245, characters 8-53
Called from file "camlinternalLazy.ml", line 27, characters 17-27
Re-raised at file "camlinternalLazy.ml", line 34, characters 4-11
Called from file "prep.ml", line 935, characters 13-26
Called from file "util.ml", line 216, characters 17-20
Called from file "prep.ml", line 1023, characters 17-47
Called from file "schedule.ml", line 85, characters 17-24
Called from file "schedule.ml", line 143, characters 37-62
Called from file "schedule.ml", line 183, characters 4-14
Called from file "tlapm.ml", line 88, characters 2-56
Called from file "tlapm.ml", line 257, characters 6-59
Called from file "tlapm.ml", line 333, characters 23-43
Called from file "list.ml", line 96, characters 24-34
Called from file "tlapm.ml", line 336, characters 13-40
Called from file "tlapm.ml", line 348, characters 8-33

version == "1.4.5 (build 33809)"
built_with == "OCaml 4.05.0"
tlapm_executable == "/home/karolis/Programs/TLAPS-1.4.5/bin/tlapm"
max_threads == 8
library_path == "/home/karolis/Programs/TLAPS-1.4.5/lib/tlaps"
search_path == << "/home/karolis/Programs/TLAPS-1.4.5/lib/tlaps/"
                , "/home/karolis/Programs/TLAPS-1.4.5/lib/tlaps" >>
Isabelle == "PATH='/home/karolis/.cabal/bin:/home/karolis/Programs/TLAPS-1.4.5/bin:/opt/tsung-1.6.0/bin:/home/karolis/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/karolis/Programs/TLAPS-1.4.5/bin:/home/karolis/Programs/TLAPS-1.4.5/lib/tlaps/bin'; isabelle-process -r -q -e \"(use_thy \\\"$file\\\"; writeln \\\"((TLAPS SUCCESS))\\\");\" TLA+"
Isabelle version == "Isabelle2011-1: October 2011"
zenon == "PATH='/home/karolis/.cabal/bin:/home/karolis/Programs/TLAPS-1.4.5/bin:/opt/tsung-1.6.0/bin:/home/karolis/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/karolis/Programs/TLAPS-1.4.5/bin:/home/karolis/Programs/TLAPS-1.4.5/lib/tlaps/bin'; zenon -p0 -x tla -oisar -max-time 1d \"$file\""
zenon version == "zenon version 0.8.4 [a268] 2017-11-14"
flatten_obligations == TRUE
normalize == TRUE
```
Here is the spec, that causes the problem (Ctrl+G, Ctrl+G on line 131: <4>2. Cardinality(l) \in Nat BY <2>l, FS_CardinalityType):
```
---------------------------- MODULE AxbLimitConq ----------------------------
EXTENDS Naturals, FiniteSets, TLAPS, FiniteSetTheorems, NaturalsInduction, WellFoundedInduction
CONSTANT R  \* A set of requests. 
CONSTANT M  \* A set of valid maximums.
VARIABLE m  \* Max concurrent operations.
VARIABLE p  \* Pending requests (not issued yet).
VARIABLE w  \* Waiting queue, we will ignore ordering in this spec.
VARIABLE l  \* Currently locked (being executed).
ASSUME R_Assumption == IsFiniteSet(R)
ASSUME M_Assumption == IsFiniteSet(M) /\ M \subseteq Nat /\ M # {}
THEOREM M_Props == M \subseteq Nat BY M_Assumption
vars == <<m, p, w, l>>
TypeOK ==
  /\ m \in M
  /\ p \subseteq R
  /\ w \subseteq R
  /\ l \subseteq R
Lock ==
  \E r \in p:
    /\ IF Cardinality(l) < m
       THEN UNCHANGED w /\ l' = l \cup {r}
       ELSE w' = w \cup {r} /\ UNCHANGED l
    /\ p' = p \ {r}
    /\ UNCHANGED <<m>>
UnlockWaiting ==
  \E r \in w:
    /\ w' = w \ {r}
    /\ UNCHANGED <<m, p, l>>  
UnlockLocked ==
  \E r \in l:
    \/ Cardinality(l) =< m /\ \E rr \in w:
       /\ w' = w \ {rr}
       /\ l' = (l \ {r}) \cup {rr}
       /\ UNCHANGED <<m, p>>
    \/ /\ Cardinality(l) > m \/ w = {} 
       /\ l' = l \ {r}
       /\ UNCHANGED <<m, p, w>>
Unlock == UnlockWaiting \/ UnlockLocked
Reconf ==   \* TODO: Make the missing locks here?
  /\ m' \in M
  /\ UNCHANGED <<p, w, l>>
Init ==
  /\ m \in M
  /\ p = R
  /\ w = {}
  /\ l = {}
Next == Lock \/ Unlock \/ Reconf
Spec == Init /\ [][Next]_vars
-----------------------------------------------------------------------------
NoDuplicates == \A r \in R: r \notin w \/ r \notin l
WorstCaseLimit == \E mm \in M : Cardinality(l) <= mm
THEOREM Spec => /\ []TypeOK
                /\ []NoDuplicates
                /\ []WorstCaseLimit

(************************************************************************)
(* TBD: If limit is reached, the locked set can only decrease,          *)
(* unless the limit is increased.                                       *)
(************************************************************************)
LimitingWorks ==
  Cardinality(l) > m => 
    \/ Cardinality(l') < Cardinality(l)
    \/ m' >= Cardinality(l) 

-----------------------------------------------------------------------------
THEOREM SpecTypeOK == Spec => TypeOK
  <1> USE DEF TypeOK, vars
  <1>1. Init => TypeOK BY DEF Init
  <1>2. TypeOK /\ [Next]_vars => TypeOK'
    <2> SUFFICES ASSUME TypeOK, Next PROVE TypeOK' OBVIOUS
    <2> QED BY DEF Next, Lock, Unlock, UnlockWaiting, UnlockLocked, Reconf
  <1> QED BY <1>1, <1>2, PTL DEF Spec, vars

-----------------------------------------------------------------------------
LEMMA FiniteNatSubsetHasMax ==
  \A S : IsFiniteSet(S) /\ S \subseteq Nat /\ S # {} => \E x \in S : \A y \in S : x \geq y
  <1> DEFINE FNSE(S) == S \subseteq Nat /\ S # {} => \E x \in S : \A y \in S : x \geq y
  <1> HIDE DEF FNSE
  <1> SUFFICES ASSUME TRUE
               PROVE \A S : IsFiniteSet(S) => FNSE(S)
      BY DEF FNSE
  <1>b. ASSUME NEW X, X = {}
        PROVE FNSE(X)
        BY <1>b DEF FNSE
  <1>i. ASSUME NEW X, IsFiniteSet(X), FNSE(X), NEW x, x \notin X
        PROVE  FNSE(X \cup {x})
        <2> SUFFICES ASSUME X \cup {x} \subseteq Nat
                     PROVE \E xx \in X \cup {x} : \A yy \in X \cup {x} : xx \geq yy
            BY <1>i DEF FNSE 
        <2>1. CASE \A s \in X: x > s BY <2>1, <1>i
        <2>2. CASE \E s \in X: s > x BY <2>2, <1>i DEF FNSE
        <2> QED BY <2>1, <2>2, <1>i
  <1> QED BY ONLY <1>b, <1>i, FS_Induction, IsaM("blast")

-----------------------------------------------------------------------------
LEMMA M_Transitive == \A a, b, c \in M : a =< b /\ b =< c => a =< c
  BY M_Assumption

-----------------------------------------------------------------------------
THEOREM SpecWorstCaseLimit == Spec => WorstCaseLimit
  <1> USE DEF WorstCaseLimit, vars
  <1>1. Init => WorstCaseLimit BY M_Assumption, FS_EmptySet DEF Init
  <1>2. WorstCaseLimit /\ TypeOK /\ [Next]_vars => WorstCaseLimit'
    <2> SUFFICES ASSUME WorstCaseLimit, Next, TypeOK PROVE WorstCaseLimit' OBVIOUS
    <2> USE DEF Next
    <2> PICK gm \in M : \A om \in M : gm \geq om BY FiniteNatSubsetHasMax, M_Assumption
    <2> m =< gm /\ gm \in M /\ \A mm \in M : gm >= mm BY DEF TypeOK
    <2>l. IsFiniteSet(l) BY R_Assumption, FS_Subset DEF TypeOK
    <2> Cardinality(l) =< gm
       <3>1. PICK mm \in M : Cardinality(l) =< mm OBVIOUS
       <3>2. mm =< gm OBVIOUS
       \*<3>3. IsFiniteSet(l) BY R_Assumption, FS_Subset DEF TypeOK
       <3>4. mm \in Nat /\ gm \in Nat BY <3>1, M_Assumption
       <3>q. QED BY ONLY <3>1, <3>2, <2>l, <3>4, FS_CardinalityType
    <2> WITNESS gm \in M \* Target: (Cardinality(l) =< gm)'
    <2>1. CASE Lock \* BY <2>1 DEF Lock
      \* WorstCaseLimit == \E mm \in M : Cardinality(l) <= mm
      \* Lock ==
      \*   \E r \in p:
      \*     /\ IF Cardinality(l) < m
      \*        THEN UNCHANGED w /\ l' = l \cup {r}
      \*        ELSE w' = w \cup {r} /\ UNCHANGED l
      \*     /\ p' = p \ {r}
      \*     /\ UNCHANGED <<m>>
      
      <3> PICK rr \in p : Lock!(rr) BY <2>1 DEF Lock
      <3>a. CASE Cardinality(l) <  m BY <3>a, <2>1 DEF Lock, TypeOK
      <3>b. CASE Cardinality(l) >= m \*BY <3>b, <2>1 DEF Lock
        <4>1. m \in Nat BY M_Props DEF TypeOK
        \*<4>x. IsFiniteSet(l) BY R_Assumption, FS_Subset DEF TypeOK
        <4>2. Cardinality(l) \in Nat BY <2>l, FS_CardinalityType
        <4>3. l' = l BY ONLY <4>1, <3>b, <2>1 DEF Lock
        <4>q. QED
      <3>q. QED BY <3>a, <3>b
      
      
      (*
      <3> \E xm \in M : TRUE BY M_Assumption
      <3> \E xm \in M : \A xmm \in M : TRUE BY M_Assumption
      <3> \A a, b \in Nat: a \leq b \/ a \geq b OBVIOUS
      <3> \A x \in M: x \geq 0 BY M_Assumption
      <3> \E xm \in M : \A xmm \in M : xm \geq xmm BY FiniteNatSubsetHasMax, M_Assumption
      <3> \A xmm \in M : xmm <= gm OBVIOUS
      <3> m \in M BY DEF TypeOK
      <3> m =< gm BY DEF TypeOK
      *)
      (*
        <4> DEFINE HasMin(S) == \E xm \in S : \A xmm \in S : xm \leq xmm
        <4>b. ASSUME NEW mm \in Nat, M = {mm}
              PROVE HasMin(M) \*\E xm \in M : \A xmm \in M : xm \leq xmm
              BY ONLY <4>b
        <4>i. ASSUME M \subseteq Nat,
                     HasMin(M),
                     NEW mm \in Nat
              PROVE HasMin(M \cup {mm})
        <4> HIDE DEF HasMin
        <4> QED BY ONLY <4>b, <4>i \*, M_Assumption, WFInduction, IsWellFoundedOnSubset
      *)
      (*
          \* <4> PICK xm : \A xmm \in M : xm \leq xmm BY M_Assumption
          <4> PICK xm \in M : TRUE BY M_Assumption
          <4> PICK ym \in M : ym \in M BY M_Assumption
          \* <4> WITNESS xm \in M
          <4> QED OBVIOUS
      *)
      \*  BY ONLY M_Assumption \*, NatInduction, SimpleArithmetic, SetExtensionality, NoSetContainsEverything
      \* <3> PICK r : Lock!(r) BY <2>1 DEF Lock
    <2>2. CASE UnlockWaiting BY <2>2 DEF UnlockWaiting
    <2>3. CASE UnlockLocked \* BY <2>3 DEF UnlockLocked
    <2>4. CASE Reconf BY <2>4 DEF Reconf
    <2> QED BY <2>1, <2>2, <2>3, <2>4 DEF Unlock
  <1> QED BY <1>1, <1>2, PTL, SpecTypeOK DEF Spec, vars

-----------------------------------------------------------------------------
XA == TRUE
XB == TRUE
XC == {1, 2, 3}
XD(c) == c = 1 /\ (XA \/ XB)
THEOREM XA \/ XB => \E c \in XC : XD(c)
  <1> SUFFICES ASSUME XA \/ XB PROVE \E c \in XC : XD(c) OBVIOUS
  <1> 1 \in XC BY DEF XC
  <1> WITNESS 1 \in XC
  <1>a. CASE XA \* BY DEF XD
  <1>b. CASE XB \* BY DEF XD
  <1> QED BY <1>a, <1>b

THEOREM ASSUME NEW a \in Nat,
               NEW b \in Nat,
               NEW s
        PROVE Cardinality(s) =< b
  <1>1. Cardinality(s) =< a OMITTED
  <1>2. a =< b              OMITTED
  <1> QED BY <1>1, <1>2
THEOREM ASSUME NEW a \in Nat,
               NEW b \in Nat,
               NEW s, IsFiniteSet(s)
        PROVE Cardinality(s) =< b
  <1>1. Cardinality(s) =< a OMITTED
  <1>2. a =< b              OMITTED
  <1> QED BY <1>1, <1>2, FS_CardinalityType

=============================================================================
```
The spec itself is incomplete experiment, although I was unable to reproduce a bug in a smaller spec.

**Comment by kape1395** [comment]
Forgot to mention, versions:
TLA Toolbox: 1.7.0 
TLAPM: 1.4.5

Another interesting observation: if the same line `Cardinality(l) \in Nat BY <2>lf, FS_CardinalityType` is moved up to the <2> level, the step is proven successfully. E.g:
```
    <2> m =< gm /\ gm \in M /\ \A mm \in M : gm >= mm BY DEF TypeOK
    <2>lf. IsFiniteSet(l) BY R_Assumption, FS_Subset DEF TypeOK
    <2>lc. Cardinality(l) \in Nat BY <2>lf, FS_CardinalityType
```

**Comment by quicquid** [comment]
Hello! I could reproduce the crash on my machine and pinpoint the issue to this obligation:

    <3> PICK rr \in p : Lock!(rr) BY <2>1 DEF Lock

If that one is commented out, the PM does not crash any more.

**Comment by johnyf** [comment]
The assertion error reported above (https://github.com/tlaplus/tlapm/issues/8#issue-651129831), has been addressed as of commit 3f0fad001efa79948fbdca7000118489648791aa, as anticipated from the comment https://github.com/tlaplus/tlapm/pull/9#issuecomment-655787094, because #9 has been merged.

(`tlapm` does not prove 3 obligations from the module `AxbLimitConq` given above (https://github.com/tlaplus/tlapm/issues/8#issue-651129831), but this issue is about the assertion error, which did happened before any proving occurred.)

`tlapm` proves the step `<4>2. Cardinality(l) \in Nat BY <2>l, FS_CardinalityType` of `AxbLimitConq` (related to comment https://github.com/tlaplus/tlapm/issues/8#issuecomment-653930600).
